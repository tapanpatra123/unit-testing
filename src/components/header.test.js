import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import Header from './header'

describe("header section",()=>{

    let wrapper;
    beforeEach(()=>{
        wrapper = shallow(<Header />)
    })

    it('to snapshot',()=>{
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

    it("render without error",()=>{
        const header = wrapper.find(`[data-list="headerSection"]`)
        expect(header.length).toBe(1)
    })

    it("display logo", ()=>{
        const logo = wrapper.find(`[data-list="HeaderLogo"]`)
        expect(logo.length).toBe(1)
    })
})