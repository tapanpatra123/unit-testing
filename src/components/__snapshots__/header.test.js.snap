// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header section to snapshot 1`] = `
<header
  data-list="headerSection"
>
  <img
    data-list="HeaderLogo"
    src="https://respinor.com/wp-content/uploads/2017/04/logo-dummy.png"
  />
</header>
`;
